### PR TITLE
Add sshpass recipe

### PR DESCRIPTION
**Pull request by asafkahlon**
Signed-off-by: Asaf Kahlon <asafka7@gmail.com>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/sshpass/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - make
+    - pkgconfig
+
+test:
+  commands:
+    - test -x ${PREFIX}/bin/sshpass
+
+about:
+  home: https://sourceforge.net/projects/sshpass/
+  license: GPL-2.0+
```
Comment (chrisburr) [inline recipes/sshpass/meta.yaml]: We prefer SPDX identifiers in the hope of one day having consistent metadata everywhere :)
```suggestion
  license: GPL-2.0-or-later
```

File: recipes/sshpass/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - make
+    - pkgconfig
+
+test:
+  commands:
+    - test -x ${PREFIX}/bin/sshpass
```
Comment (chrisburr) [inline recipes/sshpass/meta.yaml]: Is there a `sshpass --help` or `sshpass --version` you can run?

File: recipes/sshpass/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - ${PREFIX}/bin/sshpass -h
```
Comment (chrisburr) [inline recipes/sshpass/meta.yaml]: No need for the prefix as it should be on `$PATH`
```suggestion
    - sshpass -h
```

Comment (asafkahlon) [reply recipes/sshpass/meta.yaml]: You're right, fixed.

**Comment by asafkahlon** [comment]
@conda-forge/staged-recipes Anything else?

**Pull request merged**